WordPress/gutenberg
# Fix sticky position in classic themes with appearance tools support

## What?


Fixes one of the items in #56131: Extra wrapper on wide/full aligned blocks interferes with sticky position styles

Sticky position works in relation to its parent container, so when its classnames are added to a block wrapped in a legacy `data-align` wrapper, it doesn't work. This fixes the issue by detecting the sticky classes and adding them to the align wrapper instead.



## Testing Instructions





1. Add support for appearance tools to your classic theme of choice by pasting `add_theme_support( 'appearance-tools' );` into the themes's setup function.
2. In the editor, add a Group block, set its alignment to "wide" (in the block toolbar) and its position to "sticky" (in the block settings sidebar). You may need to add some more blocks to the page content in order for it to be scrollable.
3. Check that the block actually sticks to the top of the viewport when scrolling.
4. Switch to a block theme and check that the block is still sticky in the editor.
